pkg/sdk/tasks_gen_integration_test.go

```diff
@@ -0,0 +1,509 @@
+		err = client.Tasks.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+
+		alteredTask, err := client.Tasks.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, "started", alteredTask.State)
+
+		alterRequest = NewAlterTaskRequest(id).WithSuspend(Bool(true))
+		err = client.Tasks.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+
+		alteredTask, err = client.Tasks.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, "suspended", alteredTask.State)
+	})
+
+	t.Run("alter task: remove after and add after", func(t *testing.T) {
+		otherName := randomString(t)
+		otherId := NewSchemaObjectIdentifier(database.Name, schema.Name, otherName)
+
+		request := NewCreateTaskRequest(otherId, sql).WithSchedule(String("10 MINUTE"))
+
+		err := client.Tasks.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupTaskProvider(otherId))
+
+		name := randomString(t)
+		id := NewSchemaObjectIdentifier(database.Name, schema.Name, name)
+
+		request = NewCreateTaskRequest(id, sql).
+			WithAfter([]SchemaObjectIdentifier{otherId})
+
+		err = client.Tasks.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupTaskProvider(id))
+
+		task, err := client.Tasks.ShowByID(ctx, id)
+
+		require.NoError(t, err)
+		assert.Contains(t, task.Predecessors, otherId.Name())
+
+		alterRequest := NewAlterTaskRequest(id).WithRemoveAfter([]SchemaObjectIdentifier{otherId})
+
+		err = client.Tasks.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+
+		task, err = client.Tasks.ShowByID(ctx, id)
+
+		require.NoError(t, err)
+		assert.Equal(t, "[]", task.Predecessors)
+
+		alterRequest = NewAlterTaskRequest(id).WithAddAfter([]SchemaObjectIdentifier{otherId})
+
+		err = client.Tasks.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+
+		task, err = client.Tasks.ShowByID(ctx, id)
+
+		require.NoError(t, err)
+		assert.Contains(t, task.Predecessors, otherId.Name())
+	})
+
+	t.Run("alter task: modify when and as", func(t *testing.T) {
+		task := createTask(t)
+		id := task.ID()
+
+		newSql := "SELECT CURRENT_DATE"
+		alterRequest := NewAlterTaskRequest(id).WithModifyAs(String(newSql))
+		err := client.Tasks.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+
+		alteredTask, err := client.Tasks.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, newSql, alteredTask.Definition)
+
+		newWhen := `SYSTEM$STREAM_HAS_DATA('MYSTREAM')`
+		alterRequest = NewAlterTaskRequest(id).WithModifyWhen(String(newWhen))
+		err = client.Tasks.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+
+		alteredTask, err = client.Tasks.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, newWhen, alteredTask.Condition)
+	})
+
+	t.Run("show task: default", func(t *testing.T) {
+		task1 := createTask(t)
+		task2 := createTask(t)
+
+		showRequest := NewShowTaskRequest()
+		returnedTasks, err := client.Tasks.Show(ctx, showRequest)
+		require.NoError(t, err)
+
+		assert.Equal(t, 2, len(returnedTasks))
```

Comment:
Yeah, maybe I forgot and actually thought about two parallel tests case that you mentioned, but you're right that might not be the case here, because Tasks are not as common as creating databases schemas etc.